src/openrct2/ride/Track.cpp

```diff
@@ -774,7 +774,7 @@ std::optional<CoordsXYZD> GetTrackSegmentOrigin(const CoordsXYE& posEl)
     const auto& trackBlock = ted.sequences[sequenceIndex].clearance;
     CoordsXY trackBlockOffset = { trackBlock.x, trackBlock.y };
     coords += trackBlockOffset.Rotate(DirectionReverse(direction));
-    coords.z -= trackBlock.z;
+    coords.z += ted.sequences[0].clearance.z - trackBlock.z;
 
     return CoordsXYZD(coords, direction);
 }
```

Comment:
It looks like `GetTrackSegmentOrigin()` is only called from `ScTrackIterator.cpp`, so maybe we should move it there. (If another team member agrees that is, don’t move it just yet.)

Reply:
I said the exact same thing in the original pr

Reply:
@duncanspumpkin Looking back at the coversation of the previous PR, wasn't that about `Vehicle::UpdateTrackChange`?

As for `GetTrackSegmentOrigin()`, it was put there originally by IntelOrca 3 years ago. I'm suspecting it was put there to be used as a more general purpose function? As right now this functionality of getting a track piece origin is copy/pasted/duplicated around the code base quite a lot.

For now, I can move either or both functions, in either this PR or another, whichever is preferred? 🙂

Reply:
Do the other places employ the same `coords.z += ted.sequences[0].clearance.z - trackBlock.z;` thing?

Reply:
It's done in a [variety of ways](https://github.com/search?q=repo%3AOpenRCT2%2FOpenRCT2+sequences%5B0%5D.clearance&type=code), but the one I specifically looked at is this one:
https://github.com/OpenRCT2/OpenRCT2/blob/21001954674b2e4a8b4973f44c5b51905aa9302d/src/openrct2/ride/RideConstruction.cpp#L412-L414

Which is called when you right-click a track element in construction mode (which made it easy to check the flow on various track pieces).

Reply:
Let’s keep it where it is for now.